common/coin_support/coin_utils.h

```diff
@@ -132,6 +148,7 @@ typedef struct
     uint64_t network_chain_id;
 
     uint8_t is_harmony_address;
+    uint16_t address_tag;
```

Comment:
Is the `address_tag` a generic variable that could be used in other chain paths too?
If `address_tag` is only useful for solana, then we should rename it as `sol_address_tag`.

We could also have a union variable of size 64 bytes or more, which could be used to populate any chain specific variables like `address_tag` and `is_harmony_address` and not needing to update the txn_metadata.

Reply:
this is going to be generic. not specific to solana. currently its defined only for solana. but can be used for others as well.